attachment_db_by_checksum/models/ir_attachment.py

```diff
@@ -0,0 +1,72 @@
+import logging
+from odoo import models, api, _
+from odoo.exceptions import AccessError
+
+_logger = logging.getLogger(__name__)
+
+
+class Attachment(models.Model):
+    _inherit = "ir.attachment"
+
+    @api.model
+    def _file_write(self, value, checksum):
+        location = self._storage()
+        if location != "hashed_db":
+            return super(Attachment, self)._file_write(value, checksum)
+        fname, _ = self._get_path(False, checksum)
+        att = self.env["ir.attachment.content"].search([
+            ("checksum", "=", fname)
```

Comment:
```suggestion
            ("checksum", "=", fname), limit=1
```
same for all similar search (which could be delegated to `_find_attachment_content`

Reply:
Done